docs/source/nodes_and_pipelines/namespaces.md

```diff
@@ -0,0 +1,336 @@
+from .nodes import evaluate_model, split_data, train_model
+
+base_data_science = pipeline(
+        [
+            node(
+                func=split_data,
+                inputs=["model_input_table", "params:model_options"],
+                outputs=["X_train", "X_test", "y_train", "y_test"],
+                name="split_data_node",
+            ),
+            node(
+                func=train_model,
+                inputs=["X_train", "y_train"],
+                outputs="regressor",
+                name="train_model_node",
+            ),
+            node(
+                func=evaluate_model,
+                inputs=["regressor", "X_test", "y_test"],
+                outputs=None,
+                name="evaluate_model_node",
+            ),
+        ]
+    )  # Creating a base data science pipeline that will be reused with different model training parameters
+
+
+def create_pipeline(**kwargs) -> Pipeline:
+    return pipeline(
+        [base_data_science],  # Creating a new data_science pipeline based on base_data_science pipeline
+        parameters={"params:model_options": "params:model_options_1"},  # Using a new set of parameters to train model
+    )
+```
+
+To use a new set of parameters, you should create a second parameters file to ovewrite parameters specified in  `conf/base/parameters.yml`. To overwrite the parameter `model_options`, create a file  `conf/base/parameters_data_science.yml` and add a parameter called `model_options_1`:
+
+```python
+#conf/base/parameters.yml
+model_options_1:
+  test_size: 0.15
+  random_state: 3
+  features:
+    - passenger_capacity
+    - crew
+    - d_check_complete
+    - moon_clearance_complete
+    - company_rating
+```
+
+> **Warning**: In Kedro, you cannot run pipelines with the same node names. In this example, both pipelines have nodes with the same names, so it's impossible to execute them together. However, `base_data_science` is not registered and will not be executed with the `kedro run` command. The `data_science` pipeline created inside the `create_pipeline()` function, based on `base_data_science`, will be executed during `kedro run` because it will be autodiscovered by Kedro.
+
+If you want to execute `base_data_science` and `data_science` pipelines together or reuse `base_data_science` a few more times, you need to modify the node names. The easiest way to do this is by using namespaces.
+
+## What is a namespace
+
+A namespace is a way to isolate nodes, inputs, outputs, and parameters inside your pipeline. If you put `namespace="namespace_name"` attribute inside the `pipeline()` creation function, it will add the `namespace_name.` prefix to all nodes, inputs, outputs, and parameters inside your new pipeline.
+
+Let's extend our previous example and try to reuse the `base_data_science` pipeline one more time by creating another pipeline based on it. First, we should use the `kedro pipeline create` command to create a new blank pipeline named `data_science_2`:
```

Comment:
Ok this is very very unclear. Why to reuse an existing pipeline that's designed to be modular you're running `kedro pipeline create`? Isn't a more intuitive flow to define this abstract, unregistered pipeline once and then initialize it's actual instances to be registed within the same `project/pipelines/pipeline` folder?

Reply:
Thank you for that proposal, @yury-fedotov. When I wrote that manual, I also thought about how I can reuse a pipeline created with `kedro pipeline create` command since the pipeline object is created inside the function. Should I create an additional pipeline with the `kedro pipeline create` command to reuse the first one? I believe it's important to provide our users with best practices, and I would be happy to hear more opinions on that. Maybe @astrojuanlu and @merelcht have some insights.

Reply:
From my point of you it doesn't matter how you create the pipelines, what's important is how you re-use them in the end. It's up to the user if they want the pipelines to be in the same folder or separated out (which is what happens when you do `kedro pipeline create`).